### ISSUE TITLE
Consider removing update check

### ISSUE THREAD

**Issue by ericmj** [rrrene/credo#403]
It seems that some broken CI scripts are spamming Hex's HTTP API. Right now requests for /api/packages/credo are about 95-99% of all requests to the API. Some of the clients also seem to be bad on bad internet connections which is causing Heroku to spam our error logs with timeouts.

Here is an example of the access logs: https://gist.github.com/ericmj/60dcc29de0f9f2aac8232ceb3e56fe9d.

I would suggest removing the update check, disabling it for the common cases of it running in CI or moving the check to another API than Hex's.

**Comment by ericmj** [comment]
The reason we separate the registry and API in Hex was so that we can offload the API and have higher reliability for package consumers. The problem is a when a tool like credo, that is very common in CI pipelines make requests directly to the API because now the API is used in those very cases we tried to use the registry instead.

**Comment by warmwaffles** [comment]
We just ran into this same problem.

```
** (FunctionClauseError) no function clause matching in Credo.CheckForUpdates.hex_versions/1
    lib/credo/check_for_updates.ex:101: Credo.CheckForUpdates.hex_versions(%{"message" => "Blocked", "status" => 403})
    lib/credo/check_for_updates.ex:16: Credo.CheckForUpdates.run/0
    lib/credo/check_for_updates.ex:7: Credo.CheckForUpdates.run/1
    lib/credo/execution/task.ex:36: Credo.Execution.Task.run/3
    (elixir) lib/enum.ex:1755: Enum."-reduce/3-lists^foldl/2-0-"/3
    (stdlib) timer.erl:166: :timer.tc/1
    lib/credo/execution/task_group.ex:52: Credo.Execution.TaskGroup.run/3
    (elixir) lib/enum.ex:1755: Enum."-reduce/3-lists^foldl/2-0-"/3
```

**Comment by ericmj** [comment]
@warmwaffles It is likely your CI server was IP blocked. Do you use a service for running CI and if so what service is it? If not can you elaborate on how your CI is set up?

**Comment by warmwaffles** [comment]
This happened on Travis. It happens intermittently.

Test script that we run is:

```sh
set -e

export MIX_ENV=test

mix local.hex --force
mix local.rebar --force

mix deps.get
mix credo

mix ecto.create
mix ecto.load

mix test
```

travis yml

```yml
language: elixir
sudo: false
dist: trusty
elixir: 1.4.2
otp_release:
  - 19.3
services:
  - redis-server
addons:
  postgresql: 9.6
```

**Comment by GeorgeKaraszi** [comment]
Just a heads up. This happens locally too.

**Comment by warmwaffles** [comment]
I find it ironic that the issue number is 403, and that is the status. :laughing:

**Comment by ericmj** [comment]
@GeorgeKaraszi If this is happening locally you should look into rate limiting whatever it you has running otherwise it is likely you will get IP banned again. During the last 8 hours your IP has made 4000 request to /api/packages/credo.

**Comment by GeorgeKaraszi** [comment]
@ericmj Okay I think I understand whats going on here. I got visual code integrated with credo and didn't know it was calling for an update each time. I guess going forward I'll need to disable this check.

**Comment by ericmj** [comment]
@GeorgeKaraszi Thanks for letting us know what was going on.

**Comment by rrrene** [comment]
@ericmj Ouch. That's not good.

I just published `v0.8.3` which should no longer run the version check for CI systems and editor integrations.

We obviously can't patch existing versions of Credo, but if you have any further ideas how to mitigate this, I am *very* happy to help. This first patch is just a start.

**Comment by warmwaffles** [comment]
We are going to upgrade, so that's one less

**Comment by warmwaffles** [comment]
@rrrene on a second thought, is the check actually necessary?

**Comment by ericmj** [comment]
@rrrene Thank you!

Personally I don't think the check is necessary, specially not every time credo is run. People that are interested in running on latest can also run `mix hex.outdated`, for people that don't care about running the latest version the check is just noise.

**Comment by rrrene** [comment]
@ericmj TBH, you are right. We should just remove this check for updates. 

It was added a long time ago when I released weekly updates and thought that a lot of people would build the escript and use Credo from the command line where they would see the hint for a new version.

This does not match today's usage of Credo. Nor was running the check every time ever a good idea to begin with.

**Comment by rrrene** [comment]
I released Credo `v0.8.4`, which completely removes the update check, shortly after the last comment, but forgot to update this issue.

I will close this now. In case there are any problems left, please feel free to re-open! :+1:
